src/OpenRoad.cc

```diff
@@ -355,15 +359,17 @@ static odb::defout::Version stringToDefVersion(const string& version)
 
 void OpenRoad::writeDef(const char* filename, const string& version)
 {
+  std::string tmp_filename = utl::createTmpFileName(filename);
   odb::dbChip* chip = db_->getChip();
   if (chip) {
     odb::dbBlock* block = chip->getBlock();
     if (block) {
       odb::defout def_writer(logger_);
       def_writer.setVersion(stringToDefVersion(version));
-      def_writer.writeBlock(block, filename);
+      def_writer.writeBlock(block, tmp_filename.c_str());
     }
   }
+  fs::rename(tmp_filename, filename);
```

Comment:
does this overwrite if the file exists?

Reply:
if the rename fails, the tmp file is left behind. intentional?

Reply:
1. The  `std::filesystem::rename` will overwrite existing file (`filename` is first deleted, then, without allowing other processes to observe `filename` as deleted, the pathname `filename` is linked to the file and `tmp_filename` is unlinked from the file.) 
2. In the initial implementation the tmp file was left behind intentionally as the tmp file was written in the `/tmp` directory, which is cleaned by the OS. With your suggested solution we will remove it.